### ISSUE TITLE
IRSA (AssumeRoleWebIdentityCredentials) requires aud(ience) claim

### ISSUE THREAD

**Issue by Patricol** [aws/aws-sdk-ruby#2698]
### Describe the bug

Aws::STS::Errors::InvalidIdentityToken "Missing a required claim: aud" exception when trying to use [AssumeRoleWebIdentityCredentials](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/AssumeRoleWebIdentityCredentials.html) to authenticate via [IRSA](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/). (Thus using Kubernetes service account token).

IRSA is not documented to require an `aud` (audience) `jwt` token field; and does not in any other implementation of it I've used.

I've checked the token; it has no `aud` field set at all; so it's not a matter of the field being present but having an incorrect value.

Kubernetes service account tokens also do not generally include `aud` values.

### Expected Behavior

`AssumeRoleWebIdentityCredentials` should not require its `web_identity_token_file` parameter contain an `aud` field.

### Current Behavior

`AssumeRoleWebIdentityCredentials` errors if its `web_identity_token_file` parameter does not contain an `aud` field.

### Reproduction Steps

Call `AssumeRoleWebIdentityCredentials`, specifying `web_identity_token_file` as the path of a kubernetes service account token mounted into the container. Potentially also setup IRSA properly. I can try to provide more detailed reproduction steps if necessary, but I'm guessing the whole "run inside kubernetes" issue makes exact reproduction cumbersome.

### Possible Solution

Don't require `aud` field in tokens used for `AssumeRoleWebIdentityCredentials`.

### Additional Information/Context

I see that 3.131.0 was released a few hours ago, I'll try with that version too.

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

aws-sdk-core 3.130.2

### Environment details (Version of Ruby, OS environment)

ruby 3.0.2p107 (2021-07-07 revision 0db68f0233), docker (debian buster) on kubernetes (ubuntu nodes)

**Comment by mullermp** [comment]
Thanks for opening an issue. The error `Aws::STS::Errors::InvalidIdentityToken` seems to be from the STS service (and not an SDK thrown error). When making an instance of `AssumeRoleWebIdentityCredentials`, can you pass in a custom STS client (using `client` option) with `http_wire_trace: true` to see what's going on under the hood? 

I haven't used this combination before, and I can do some more investigation and possibly forward to the STS team. It does look like "audience" is required here:
```
      - serviceAccountToken:
          audience: sts.amazonaws.com
          expirationSeconds: 86400
          path: token
```

**Comment by Patricol** [comment]
I also just discovered (after opening this issue) that the IRSA documentation I linked does use a token with an audience in its example. IRSA is a somewhat underdocumented feature, so this didn't show up anywhere else I looked.

I'm inclined to keep this issue open because other AWS SDKs (specifically for python) and the awscli do not require an `aud` field, and even the linked IRSA documentation doesn't describe it as a requirement, just saying that a projected token with those parameters will happen to work. My guess is that the Ruby SDK implementers took the vague documentation and interpreted it as a requirement, while other implementers did not think of it as required.

I'll try to get you that trace output.

Thanks for the quick response!
